app/src/components/avatar.rs

```diff
@@ -0,0 +1,116 @@
+use crate::infra::common_component::{CommonComponent, CommonComponentParts};
+use anyhow::{bail, Result};
+use graphql_client::GraphQLQuery;
+use yew::prelude::*;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "queries/get_user_avatar.graphql",
+    response_derives = "Debug, Hash, PartialEq, Eq, Clone",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+pub struct GetUserAvatar;
+
+pub type User = get_user_avatar::GetUserAvatarUser;
+
+pub struct Avatar {
+    common: CommonComponentParts<Self>,
+    /// The user info. If none, the error is in `error`. If `error` is None, then we haven't
+    /// received the server response yet.
+    avatar: Option<String>,
+}
+
+/// State machine describing the possible transitions of the component state.
+/// It starts out by fetching the user's details from the backend when loading.
+pub enum Msg {
+    /// Received the user details response, either the user data or an error.
+    UserAvatarResponse(Result<get_user_avatar::ResponseData>),
+    Update
+}
+
+#[derive(yew::Properties, Clone, PartialEq, Eq)]
+pub struct Props {
+    pub username: String,
+    pub width: i32,
+    pub height: i32,
+}
+
+impl CommonComponent<Avatar> for Avatar {
+    fn handle_msg(&mut self, msg: <Self as Component>::Message) -> Result<bool> {
+        match msg {
+            Msg::UserAvatarResponse(response) => match response {
+                Ok(user) => self.avatar = user.user.avatar,
+                Err(e) => {
+                    self.avatar = None;
+                    bail!("Error getting user details: {}", e);
+                }
+            },
+            Msg::Update => self.get_user_avatar(),
+        }
+        Ok(true)
+    }
+
+    fn mut_common(&mut self) -> &mut CommonComponentParts<Self> {
+        &mut self.common
+    }
+}
+
+impl Avatar {
+    fn get_user_avatar(&mut self) {
+        if self.common.username.len() > 0 {
```

Comment:
Don't use empty strings to signify "no user", but instead use an `Option<String>`

Reply:
Or just rely on the fact that you'll always be called with a valid username by construction.

Reply:
Option didn't seem to want to play nicely with Props; I think i'll just opt for the latter